### PR TITLE
fix the eventingReady status change right after the backend switch

### PR DESCRIPTION
**Pull request by VladislavPaskar**
**Description**

Changes proposed in this pull request:

- fixed the status change logic for eventPublisherReady 
- added the pods mocking into the unit tests

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/controllers/backend/reconciler_test.go [hunk 1]
```diff
@@ -442,6 +442,8 @@ var _ = Describe("Backend Reconciliation Tests", func() {
 		It("Should mark Eventing Backend CR to ready", func() {
 			ctx := context.Background()
 			natsCommander.stopErr = nil
+			ensurePublisherProxyPodIsCreated(ctx)
+			//Eventually(pod.co).Should()
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/backend/reconciler_test.go:446]: (nit) remove unused commented code.

[the remark this answers is not in the document]
Comment (VladislavPaskar) [reply components/eventing-controller/controllers/backend/reconciler_test.go]: removed the comment

Comment (VladislavPaskar) [reply components/eventing-controller/controllers/backend/reconciler_test.go]: https://github.com/kyma-project/kyma/pull/12312#discussion_r744536914

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 2]
```diff
@@ -387,6 +387,56 @@ func (r *Reconciler) UpdateBackendStatus(ctx context.Context, backendType eventi
 	return nil
 }
 
+// check the publisher proxy deployment pods for the right backend type and ready status
```
[the remark this answers is not in the document]
Comment (VladislavPaskar) [reply components/eventing-controller/controllers/backend/reconciler.go:390]: I see what you mean, but my comments are not the first in this file, which start with small letter and I honestly think we can live with that kind of in inconsistency :)

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**